packages/core/src/Asset/cip67.ts

```diff
@@ -0,0 +1,73 @@
+import { Cardano } from '..';
+import { InvalidArgumentError, OpaqueNumber } from '@cardano-sdk/util';
+import crc8 from './crc8';
+
+const ASSET_LABEL_LENGTH = 8;
+const ASSET_LABEL_BRACKET = '0';
+
+export enum AssetNameLabelNum {
+  ReferenceNFT = 100,
+  SubStandard222 = 222,
+  SubStandard333 = 333,
+  SubStandard444 = 444
+}
+
+export type AssetNameLabel = OpaqueNumber<'AssetNameLabelNum'>;
+export const AssetNameLabel = (value: number): AssetNameLabel => value as AssetNameLabel;
+
+interface DecodedAssetName {
+  label: AssetNameLabel | null;
+  content: Cardano.AssetName;
+}
+
+const assertLabelNumInterval = (labelNum: number) => {
+  if (labelNum < 0 || labelNum > 65_535) {
+    throw new InvalidArgumentError('Label num', `Label ${labelNum} out of range 0 - 65535.`);
+  }
+};
+
+const checksum = (labelNumHex: string) =>
+  crc8(Uint8Array.from(Buffer.from(labelNumHex, 'hex')))
+    .toString(16)
+    .padStart(2, '0');
+
+const isInvalidChecksum = (labelNumHex: string, labelChecksum: string) => labelChecksum !== checksum(labelNumHex);
+
+const isInvalidLength = (label: string) => label.length !== ASSET_LABEL_LENGTH;
+
+const isInvalidBracket = (lead: string, end: string) => lead !== ASSET_LABEL_BRACKET || end !== ASSET_LABEL_BRACKET;
+
+const assetLabelHexToNum = (label: string): AssetNameLabel | null => {
+  const labelLeadBracket = label[0];
+  const labelEndBracket = label[7];
+  const labelNumHex = label.slice(1, 5);
+  const labelChecksum = label.slice(5, 7);
+
+  if (
+    isInvalidLength(label) ||
+    isInvalidBracket(labelLeadBracket, labelEndBracket) ||
+    isInvalidChecksum(labelNumHex, labelChecksum)
+  ) {
+    return null;
+  }
+
+  return AssetNameLabel(Number.parseInt(labelNumHex, 16));
+};
+
+AssetNameLabel.decode = (assetId: Cardano.AssetId): DecodedAssetName => {
```

Comment:
I think we should scope this util to work with just the `AssetName`.
Also maybe let's return `null` when it fails to decode (and change the DecodedAssetName.label to non-nullable), because that indicates that it failed to decode - so there's no 'decoded asset name' in that case.
```suggestion
AssetNameLabel.decode = (assetName: Cardano.AssetName): DecodedAssetName | null => {
```